### PR TITLE
Download NuGet.exe instead of having it checked in

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @jasonmalinowski, @tannergooding, @agocke, @dotnet/roslyn-infrastructure

**Comment by tannergooding** [comment]
@jaredpar this won't work, without deleting nuget.exe, if we update the version we depend on.

**Comment by tannergooding** [comment]
Also, why don't we just download the file to he repo root?

**Comment by agocke** [comment]
@jaredpar The build task we use for restore on Unix lives in that binary. We'll have to do something similar for Unix.

**Comment by agocke** [comment]
test mac please

**Comment by jaredpar** [comment]
@tannergooding good catch. I will make the download version tolerant.  

I don't download to the root because it's bad build practice to have the build modify the source tree.  Also it will help catch any places that had a hard dependency on it being available.

**Comment by jaredpar** [comment]
@agocke doh.  Will fix.

File: Makefile [hunk 1]
```diff
@@ -9,6 +9,8 @@ BOOTSTRAP_PATH = $(BINARIES_PATH)/Bootstrap
 BUILD_LOG_PATH =
 HOME_DIR = $(shell cd ~ && pwd)
 DOTNET_VERSION = 1.0.0-preview2-002911
+NUGET_VERSION = 3.5.0-beta2
+NUGET_EXE = $(shell pwd)/nuget.exe
```
Comment (agocke) [inline Makefile]: Is this case sensitive?

Comment (jaredpar) [inline Makefile]: Yes.  The currently checked in binary has this case hence I kept it for now.

**Comment by jaredpar** [comment]
test mac please

**Comment by jaredpar** [comment]
Note: the ETA failure is expected.  I have the corresponding close change prepared and sent out a cross cutting change notification about this.

**Pull request merged**